hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSUtils.java

```diff
@@ -138,20 +137,26 @@ public static boolean isDistributedFileSystem(final FileSystem fs) throws IOExce
    * @return True if <code>pathTail</code> is tail on the path of <code>pathToSearch</code>
    */
   public static boolean isMatchingTail(final Path pathToSearch, final Path pathTail) {
-    if (pathToSearch.depth() != pathTail.depth()) return false;
     Path tailPath = pathTail;
     String tailName;
     Path toSearch = pathToSearch;
     String toSearchName;
     boolean result = false;
+
+    if (pathToSearch.depth() != pathTail.depth()) {
+      return false;
+    }
+
     do {
       tailName = tailPath.getName();
-      if (tailName == null || tailName.length() <= 0) {
```

Comment:
Does the `Path` API promise that `getName` won't return null for the versions of Hadoop in our support matrix?

Reply:
I ran 'blame' on `Path#getName()` and it says the following code has been there for 11 years now:

```
  public String getName() {
    String path = uri.getPath();
    int slash = path.lastIndexOf(SEPARATOR);
    return path.substring(slash+1);
  }
```

Reply:
if the api doesn't say it won't return null then we have to account for it possibly being null.

Reply:
`toSearchName` currently (just a few lines down) does not check for 'null' I will add there too.